core/src/main/java/org/apache/iceberg/encryption/AesGcmInputFile.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.encryption;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.io.SeekableInputStream;
+
+public class AesGcmInputFile implements InputFile {
```

Comment:
Why not implement `EncryptedInputFile`?

Reply:
`EncryptedInputFile` is an input to encryption manager's `decrypt` method; with the output being an `InputFile`, ready for direct consumption by readers. `AesGcmInputFile` is an output of the decrypt operation (its input_stream.read draws decrypted data); so it is given directly to manifest Avro readers, that use it in the usual manner (decryption is transparent to them).

Reply:
You're right. This would be the result of the `decrypt` method.